fedimint-api/src/core.rs

```diff
@@ -211,45 +242,72 @@ macro_rules! newtype_impl_display_passthrough {
 /// Static-polymorphism version of [`ModuleDecode`]
 ///
 /// All methods are static, as the decoding code is supposed to be instance-independent,
-/// at least until we start to support modules with overriden [`ModuleKey`]s
+/// at least until we start to support modules with overriden [`ModuleInstanceId`]s
 pub trait PluginDecode: Debug + Send + Sync + 'static {
+    type Input: PluginInput;
```

Comment:
We should do this change, even if we don't land the PR. Thanks to this, the user doesn't have to do this typed to erased conversion in their own code.

Reply:
Fully agree, was discussing this with @mxz42 already. Thx for including it here. This will also allow removing these types from the server and client module trait by only referring to the decoder.